app/qml/LayerList.qml

```diff
@@ -50,7 +53,7 @@ ListView {
             imageSource: iconSource ? iconSource : ""
             overlayImage: false
             highlight: highlightingAllowed && layerId === activeLayerId
-            showBorder: highlightingAllowed ? !__appSettings.defaultLayer || activeIndex - 1 !== index : true
+            showBorder: highlightingAllowed ? !__appSettings.defaultLayer || (listView.activeIndex !== index && listView.activeIndex - 1 !== index) : true
```

Comment:
This is working but it is sort of hacky and we use it all over the place. 

What if we would change it like:
 - in `ExtendedMenuItem` border would be automatically invisible when item is highlighted (no need to set `showBorder` then)
 - pass `index` and `activeIndex` to `ExtendedMenuItem` and apply highlight/border logic there

What do you think @sklencar?

Reply:
I would prefer to think about redesigning it. Borders brings a bit hack indeed, because we want not only hide borders for highlighted item, but also hide borders for an item just above. Thats because otherwise there is a weird short line above highlighted rectangular which doesn't look good.

The condition can be rewritten, of course. However, I think now its quite straightforward how item's index affects borders. Ideally, would be good to get rid of the activeIndex and use only highlight property.

For instance, one of the solutions can be to extend border line to full width and have the row element similar to MerginProjectDelegate element where borders are on full-width. Then the condition can be also simplified as suggested.

Reply:
Good points. I agree with that. 
However, it is probably not worth to dig into it now, so I am ok to merge. Good that you found it :)